### PR TITLE
Handle non JSON response

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Currently, when you try to submit a document or face capture and your wifi is disabled, the SDK does not return any meaningful error to the user because it's not able to handle non JSON responses for errors.

# Solution
Handle non JSON responses by adding a try catch statement.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [ ] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by stefaniacardenas** [comment]
@jamesstonehill In this case we are trying to handle requests with `status: 0`, for example requests that cannot be completed or when there is an error while the request is being sent. Therefore, even if we add `Accept: application/json` the response will still be `""` and `JSON.parse` would fail.

File: src/components/utils/onfidoApi.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
 import forEach from 'object-loops/for-each'
 
-const formatError = ({response, status}, onError) =>
-  onError({status, response: JSON.parse(response)})
+const formatError = ({response, status}, onError) => {
+  try {
+    onError({status, response: JSON.parse(response)})
+  }
+  catch {
+    onError({status, response: {}})
```
Comment (rfreitas) [inline src/components/utils/onfidoApi.js]: how does the upstream (e.g. popups) handle an empty error object?

Comment (stefaniacardenas) [reply src/components/utils/onfidoApi.js]: For any error which is not 401 or 422, we ignore the response object and return a generic "Connection Lost" error

Comment (rfreitas) [reply src/components/utils/onfidoApi.js]: so I'm guessing we would never get a 401 or 422 that doesn't have a json error object, right?

Comment (stefaniacardenas) [reply src/components/utils/onfidoApi.js]: Not from the API I would hope. But if for whatever reason the API returns a non json response we would not be able to map the error, therefore we would return a generic "Connection Lost" error back to the user.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-631-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://631-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**